onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc

```diff
@@ -305,92 +201,34 @@
-
-  if ("align_corners" == coordinate_mode) {
-    qnn_align_corners.bool8Value = static_cast<uint8_t>(1);
-  } else if ("half_pixel" == coordinate_mode) {
-    qnn_half_pixel.bool8Value = static_cast<uint8_t>(1);
-  }
-  QnnParamWrapper qnn_align_corners_param(node_unit.Index(), node_unit.Name(),
-                                          QNN_OP_RESIZE_BILINEAR_PARAM_ALIGN_CORNERS, qnn_align_corners);
-  QnnParamWrapper qnn_half_pixel_param(node_unit.Index(), node_unit.Name(),
-                                       QNN_OP_RESIZE_BILINEAR_PARAM_HALF_PIXEL_CENTERS, qnn_half_pixel);
-
-  std::vector<std::string> param_tensor_names;
-  param_tensor_names.push_back(qnn_align_corners_param.GetParamTensorName());
-  qnn_model_wrapper.AddParamWrapper(std::move(qnn_align_corners_param));
-  param_tensor_names.push_back(qnn_half_pixel_param.GetParamTensorName());
-  qnn_model_wrapper.AddParamWrapper(std::move(qnn_half_pixel_param));
-
-  return ProcessOutputs(qnn_model_wrapper, node_unit, std::move(input_names), std::move(param_tensor_names),
-                        logger, do_op_validation, qnn_node_type);
-}
-
-Status ResizeOpBuilder::ProcessQDQOpAttrsAndOutputs(QnnModelWrapper& qnn_model_wrapper,
-                                                    const NodeUnit& node_unit,
-                                                    std::vector<std::string>&& input_names,
-                                                    const logging::Logger& logger,
-                                                    bool do_op_validation) const {
   std::vector<std::string> param_tensor_names;
   NodeAttrHelper node_helper(node_unit);
 
   const std::string interp_mode = GetOnnxAttr(node_helper, onnx_mode_attr);
   const std::string transformation_mode = GetOnnxAttr(node_helper, onnx_coord_transf_mode_attr);
+  const std::string nearest_mode = GetOnnxAttr(node_helper, onnx_nearest_mode_attr);
+  const bool is_npu_backend = IsNpuBackend(qnn_model_wrapper.GetQnnBackendType());
   std::string qnn_op_type = "Resize";
 
-  // Handle Resize with {mode: "nearest", coordinate_transformation_mode: "asymmetric"} uniquely.
-  // QNN's ONNX converter tool translates this configuration (regardless of rounding mode)
-  // to QNN's ResizeNearestNeighbor with {align_corners: 0, half_pixel: 0}.
-  //
-  // NOTE: This is only accurate if the rounding mode is "floor". Need to investigate how to handle
-  // other rounding modes with Qualcomm. Ideally, we would use QNN's Resize operator, but it doesn't support
-  // the "asymmetric" coordinate transformation mode on HTP.
-  if (interp_mode == "nearest" && transformation_mode == "asymmetric") {
+  // Translate Resize with {mode: "nearest", nearest_mode: "floor", coordinate_transformation_mode: XXX} to
+  // QNN's ResizeNearestNeighbor operator on the HTP backend. This combination of parameters is not supported on HTP
+  // via QNN's Resize operator. Note that QNN's ResizeNearestNeighbor operator always uses "floor" rounding.
```

Comment:
Need to report this to QC.

Reply:
They already know. It's also in the QNN SDK docs.

Reply:
To clarify, the QNN SDK docs state that only the default `nearest_mode` (i.e., round_prefer_floor) is supported for QNN Resize on the HTP backend.

The docs, however, do not state that ResizeNearestNeighbor uses floor rounding. I learned that information from unit tests and from asking QC.